pyrometheus/codegen/python.py

```diff
@@ -260,7 +260,11 @@ def get_mix_molecular_weight(self, mass_fractions):
                 )
 
     def get_concentrations(self, rho, mass_fractions):
-        return self.iwts * rho * mass_fractions
+        return self._pyro_make_array([
+            % for i in range(sol.n_species):
+            self.iwts[${i}] * mass_fractions[${i}] * rho,
+            % endfor
+                ])
```

Comment:
I wonder if this will obviate mirgecom's customization of this function.  I'll test it out tomorrow.

Reply:
Hi @MTCam, following up on this. Did it obviate mirgecom's customization?

Reply:
No, it doesn't.  Not quite sure what I was thinking here.  MC squishes the concentrations below a certain level (typically ~1e-10) to 0.   The big problem is that very small mass fractions (~1e-13ish) result in significant reaction rates, which run the mass fractions significantly negative over a simulation time step.  So now we ignore everything below a certain threshold - which is set up in our customization of `get_concentrations`.